paddle/fluid/pybind/auto_parallel_py.cc

```diff
@@ -364,7 +364,30 @@ void BindAutoParallel(py::module *m) {
           py::arg("memo"))
       .def("__str__", &DeviceMesh::to_string);
 
-  py::enum_<phi::ReduceType>(*m, "ReduceType")
+  py::enum_<phi::ReduceType>(*m, "ReduceType", R"DOC(
+    Specify the type of operation used for paddle.distributed.Partial().
+    It should be one of the following values:
+
+        ReduceType.kRedSum
+        ReduceType.kRedMax
+        ReduceType.kRedMin
+        ReduceType.kRedProd
+        ReduceType.kRedAvg
+        ReduceType.kRedAny
+        ReduceType.kRedAll
```

Comment:
改成一行行的段落吧，美观点
![image](https://github.com/PaddlePaddle/Paddle/assets/70642955/120c5dd5-dd0a-448f-b43d-5680aba11497)

```suggestion
        - ReduceType.kRedSum
        - ReduceType.kRedMax
        - ReduceType.kRedMin
        - ReduceType.kRedProd
        - ReduceType.kRedAvg
        - ReduceType.kRedAny
        - ReduceType.kRedAll
```

Reply:
DONE